src/curl.c

```diff
@@ -129,6 +234,13 @@ double swupd_query_url_content_size(char *url)
 		return -1;
 	}
 
+	if (capath) {
```

Comment:
that code "picks" a usable capath. this one actually uses it.

Reply:
So at 145 we pick a usable path. What unsets the choice of the path that we need to set it again here? I see "c=curl_easy_duphandle(curl)" and think that this implies changes to c will also affect curl, so a working path will call curl_easy_init at line 71 in swupd_curl_init(), set the CA path at 146 in one of the loops, test it works at 152, then return at 176. Of course this depends on my understanding of the dup. If I am wrong, then why are we calling curl_easy_duphandle at all?

Reply:
`c` is a comletely independent CURL handle. i could well use one `curl_easy_init()`. duplication only to make sure we actually take the right starting point if it was manipulated before (e.g. if `swupd_curl_init()` were to set any opts).

so code in `swupd_curl_check_network()` creates a dup and then destroys it. i would set CApath there (on file-scope `curl`), but the rest of the code is organized so that it constantly `curl_easy_reset()` and then re-configures before calling `curl_easy_perform()`. so it would not "stick" on global since any reset would wipe it.

Reply:
ok, thanks. that makes sense.